azure-sink-connector/src/integration-test/java/io/aiven/kafka/connect/azure/AvroIntegrationTest.java

```diff
@@ -0,0 +1,357 @@
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import java.util.zip.GZIPInputStream;
+
+import org.apache.kafka.clients.producer.ProducerConfig;
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import io.aiven.kafka.connect.common.config.CompressionType;
+
+import org.apache.avro.Schema;
+import org.apache.avro.file.DataFileReader;
+import org.apache.avro.file.SeekableByteArrayInput;
+import org.apache.avro.file.SeekableInput;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericDatumReader;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.avro.util.Utf8;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
+
+@SuppressWarnings("PMD.AvoidInstantiatingObjectsInLoops")
+@Testcontainers
+final class AvroIntegrationTest extends AbstractIntegrationTest<String, GenericRecord> {
+    private static final String CONNECTOR_NAME = "aiven-azure-sink-connector-avro";
+
+    @Container
+    private final SchemaRegistryContainer schemaRegistry = new SchemaRegistryContainer(KAFKA);
+
+    private final Schema avroInputDataSchema = new Schema.Parser().parse(
+            "{\"type\":\"record\",\"name\":\"input_data\"," + "\"fields\":[{\"name\":\"name\",\"type\":\"string\"}]}");
+
+    @BeforeEach
+    void setUp() throws ExecutionException, InterruptedException {
+        testBlobAccessor.clear(azurePrefix);
+        final Map<String, Object> producerProps = new HashMap<>();
+        producerProps.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, KAFKA.getBootstrapServers());
+        producerProps.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                "io.confluent.kafka.serializers.KafkaAvroSerializer");
+        producerProps.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                "io.confluent.kafka.serializers.KafkaAvroSerializer");
+        producerProps.put("schema.registry.url", schemaRegistry.getSchemaRegistryUrl());
+        startConnectRunner(producerProps);
+    }
+
+    private void produceRecords(final int recordCountPerPartition) throws ExecutionException, InterruptedException {
+        final List<Future<RecordMetadata>> sendFutures = new ArrayList<>();
+        int cnt = 0;
+        for (int i = 0; i < recordCountPerPartition; i++) {
+            for (int partition = 0; partition < 4; partition++) {
+                final String key = "key-" + cnt;
+                final GenericRecord value = new GenericData.Record(avroInputDataSchema);
```

Comment:
We can try instantiating this out of the loop

Reply:
Not sure I fully understand what you mean. Are you suggesting reusing the record? Now we are creating `partition * recordPerPartition` records to send. If we will reuse the record object how it's value will look like?

Reply:
You are correct. Ignore this one and next one too,